### PR TITLE
#164598307 modularise the route controller

### PR DESCRIPTION
**Pull request by Eazybee**
#### What does this PR do?
- modularise the route controller

#### Description of Task to be completed?
- modularise the route controller based on class

#### Any background context you want to provide?
- You need node and npm installed
- Run ```npm install``` to install all dependencies 

#### What are the relevant pivotal tracker stories?
- #164598307  (https://www.pivotaltracker.com/story/show/164598307)

### CODE REVIEW

File: SERVER/v1/src/controller/MessageController.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    const savedMail = MessageController.savedMail(req, res);
+
+    // Send the mail
+    req.body.id = savedMail.data[0].id;
+    const sentMail = MessageController.sendDraftUtility(req, res);
+    res.status(201).json(sentMail);
+  }
+
+  static saveDraft(req, res) {
+    res.status(201).send(MessageController.savedMail(req));
```
Comment (jamesenejo) [inline SERVER/v1/src/controller/MessageController.js]: Always keep things explicit in your code. Let's rewrite this
```
const savedMail = MessageController.savedMail(req);
return res.status(201).json(savedMail);
```

File: SERVER/v1/src/controller/UserController.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import User from '../model/User';
+
+dotenv.config();
+
+class UserController {
+  static signUp(req, res) {
+    const newUser = new User(req.body.email,
+      req.body.firstName, req.body.lastName, req.body.password);
+    users.push(newUser);
+    return jwt.sign({ newUser }, process.env.JWT_PRIVATE_SECRET, (err, token) => {
```
Comment (jamesenejo) [inline SERVER/v1/src/controller/UserController.js]: I think you can refactor here. Why not have a function that creates the jwt token then use that function in both places (line 13 and line 26)?

We should always avoid duplicating our code.

File: SERVER/v1/src/controller/UserController.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+          token,
+        }],
+      });
+    });
+  }
+
+  static login(req, res) {
+    const { user } = req;
+    UserController.user = user;
+    jwt.sign({ user }, process.env.JWT_PRIVATE_SECRET, (err, token) => {
```
Comment (jamesenejo) [inline SERVER/v1/src/controller/UserController.js]: Same duplication comment apply here.

File: SERVER/v1/src/controller/UserController.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      res.status(200).json({
+        status: 200,
+        data: [{
+          token,
+        }],
+      });
+    });
+  }
+}
+UserController.user = null;
```
Comment (jamesenejo) [inline SERVER/v1/src/controller/UserController.js]: Please explain why did you do this.

File: SERVER/v1/src/controller/UserController.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+UserController.token = null;
```
Comment (jamesenejo) [inline SERVER/v1/src/controller/UserController.js]: And this...

**Review by jamesenejo — CHANGES_REQUESTED** [review]

File: SERVER/v1/src/controller/MessageController.js (hunk 1 above)
Comment (Eazybee) [reply SERVER/v1/src/controller/MessageController.js]: corrected

File: SERVER/v1/src/controller/MessageController.js [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+        subject: mail.getSubject(),
+        message: mail.getMessage(),
+        parentMessageId: mail.getParentMessageId(),
+        status: mail.getStatus(),
+      }],
+    };
+  }
+
+  static sendDraft(req, res) {
+    res.status(201).json(MessageController.sendDraftUtility(req));
```
[the remark this answers is not in the document]
Comment (Eazybee) [reply SERVER/v1/src/controller/MessageController.js]: corrected

File: SERVER/v1/src/controller/MessageController.js [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+  }
+
+  static sendDraftUtility(req) {
+    const mail = Message.getMails(parseInt(req.body.id, 10))[0];
```
[the remark this answers is not in the document]
Comment (Eazybee) [reply SERVER/v1/src/controller/MessageController.js]: corrected

File: SERVER/v1/src/controller/UserController.js (hunk 2 above)
Comment (Eazybee) [reply SERVER/v1/src/controller/UserController.js]: corrected

File: SERVER/v1/src/controller/UserController.js (hunk 3 above)
Comment (Eazybee) [reply SERVER/v1/src/controller/UserController.js]: corrected

File: SERVER/v1/src/controller/UserController.js (hunk 4 above)
Comment (Eazybee) [reply SERVER/v1/src/controller/UserController.js]: corrected

File: SERVER/v1/src/controller/UserController.js (hunk 5 above)
Comment (Eazybee) [reply SERVER/v1/src/controller/UserController.js]: corrected

**Pull request merged**